### PR TITLE
New indexing model: fix compile issues in benchmarks

### PR DESCRIPTION
**Pull request by shawnce**
@gribozavr

### CODE REVIEW

File: benchmark/single-source/RangeAssignment.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ import TestsUtils
 
 @inline(never)
 public func run_RangeAssignment(scale: Int) {
-  let range = 100..<200
+  let range = 100..<200 as Range
```
Comment (gribozavr) [inline benchmark/single-source/RangeAssignment.swift]: I think we prefer `let range: Range = ...`

File: benchmark/single-source/StringInterpolation.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ public func run_StringInterpolation(N: Int) {
       // FIXME: if String is not stored as UTF-16 on this platform, then the
       // following operation has a non-trivial cost and needs to be replaced
       // with an operation on the native storage type.
-      result = result &+ Int(s.utf16[s.utf16.endIndex.predecessor()])
+      result = result &+ Int(s.utf16[s.utf16.next(s.utf16.endIndex)])
```
Comment (gribozavr) [inline benchmark/single-source/StringInterpolation.swift]: `previous()`?

File: stdlib/public/core/CollectionAlgorithms.swift.gyb [hunk 3]
```diff
@@ -308,22 +308,6 @@ extension MutableCollection
...
-
-  @available(*, unavailable, renamed: "sorted(isOrderedBefore:)")
-  public func sort(
-    @noescape isOrderedBefore: (Iterator.Element, Iterator.Element) -> Bool
-  ) -> [Iterator.Element] {
-    fatalError("unavailable function can't be called")
-  }
-}
-%end
-
```
Comment (gribozavr) [inline stdlib/public/core/CollectionAlgorithms.swift.gyb]: This part LGTM.

**Comment by gribozavr** [comment]
> I didn't pull s.utf16 out into a let here because I wasn't sure if that was in conflict with the benchmark or not

I think it would only improve precision of the benchmark.  The line that you modified is used as a way to force the string to be evaluated.  The less work it does, the less it distorts the benchmark results.

File: benchmark/single-source/StringInterpolation.swift (hunk 2 above)
Comment (shawnce) [inline benchmark/single-source/StringInterpolation.swift]: ugh yeah... thanks

File: benchmark/single-source/RangeAssignment.swift (hunk 1 above)
Comment (shawnce) [inline benchmark/single-source/RangeAssignment.swift]: ok

**Pull request merged**